File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.kt
class Solution {
    fun firstUniqChar(s: String): Int {
        val arr = Array(26) {0}
        for (c in s) {
            arr[c-'a']++
        }
        for (i in s.indices) {
            val c = s[i]
            if (arr[c-'a'] == 1) {
                return i
            } 
        }
        return -1
    }
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.rs
impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut arr = [0;26];
        for c in s.bytes() {
            arr[(c - b'a') as usize] += 1;
        }
        for (i,c) in s.bytes().enumerate() {
            if arr[(c-b'a') as usize] == 1 {
                return i as i32;
            }
        }
        -1
    }
}
